samples/packages/openapi-spec-loader/src/main/kotlin/OpenAPISpecLoader.kt

```diff
@@ -46,24 +46,83 @@ object OpenAPISpecLoader {
                 exitProcess(4)
             }
 
-            val sourceUrl =
-                when (ctx.config.importType) {
-                    "CLOUD" -> {
-                        Utils.getInputFile(
-                            ctx.config.specFile,
-                            outputDirectory,
-                            false,
-                            ctx.config.specPrefix,
-                            ctx.config.specKey,
-                        )
+            try {
+                val sourceFiles =
+                    when (ctx.config.importType) {
+                        "DIRECT" -> listOf(ctx.config.specFile)
+                        "CLOUD" ->
+                            when {
+                                ctx.config.specKey.isBlank() && ctx.config.specPrefix.isNotBlank() -> {
+                                    Utils.getInputFiles(
+                                        ctx.config.specFile,
+                                        outputDirectory,
+                                        false,
+                                        ctx.config.specPrefix,
+                                    )
+                                }
+                                else -> {
+                                    listOf(
+                                        Utils.getInputFile(
+                                            ctx.config.specFile,
+                                            outputDirectory,
+                                            false,
+                                            ctx.config.specPrefix,
+                                            ctx.config.specKey,
+                                        ),
+                                    )
+                                }
+                            }
+                        "URL" -> listOf(ctx.config.specUrl)
+                        else -> {
+                            logger.error { "Unsupported import type: ${ctx.config.importType}" }
+                            exitProcess(5)
+                        }
                     }
-
-                    "DIRECT" -> ctx.config.specFile
-                    else -> ctx.config.specUrl
+                for (sourceFile in sourceFiles) {
+                    processFile(ctx, connectionQN, sourceFile, batchSize)
                 }
+            } catch (e: Exception) {
+                logger.error(e) { "An error occurred during OpenAPI spec processing." }
+                exitProcess(1)
+            }
```

Comment:
This full-wrapping in a try-catch should be unnecessary — in the next release I'll be adding an uncaught-exception handler which will in effect handle such scenarios automatically across all Kotlin-based custom packages.